### PR TITLE
[changelog-generator] Accept "Changelog:All" and inline entries

### PR DESCRIPTION
**Pull request by lluiscampos**
This adds to changes to changelog-generator to accommodate commits by
dependabot, which use titles like "Changelog:All : Important info here"

See new entry in the test for better understanding.

### CODE REVIEW

**Comment by lluiscampos** [comment]
Stopping the pipeline, this tool is not used there.

**Comment by lluiscampos** [comment]
See also https://github.com/mendersoftware/mendertesting/pull/84

File: extra/changelog-generator/changelog-generator [hunk 1]
```diff
@@ -148,14 +148,15 @@ for repo in repos:
...
-            if match:
+            match_changelog = re.match("^ *Changelog: *(.*)", line, re.IGNORECASE)
+            if match_changelog:
                 if log_entry:
                     add_entry(sha, log_entry)
                     log_entry = ""
                 log_entry_local = False
 
-                if re.match("^(Title|Commit|None)[ .]*$", match.group(1), re.IGNORECASE):
+                match_keyword = re.match("^(Title|Commit|All|None)[ .:]*(.*)$", match_changelog.group(1), re.IGNORECASE)
```
Comment (kacf) [inline extra/changelog-generator/changelog-generator:158]: Unless we really can't because of dependabot, I think we should require colon after the tag, otherwise this entry will be misinterpreted:

```
Changelog: All crashing bugs fixed
```

You might even consider a test case for this.

File: extra/changelog-generator/changelog-generator [hunk 2]
```diff
@@ -166,15 +167,20 @@ for repo in repos:
                                                  % sha)
                     exclusive_tag_seen = True
 
-                if re.match("^Title[ .]*$", match.group(1), re.IGNORECASE):
-                    log_entry = title
-                elif re.match("^Commit[ .]*$", match.group(1), re.IGNORECASE):
-                    log_entry_commit = True
-                elif re.match("^None[ .]*$", match.group(1), re.IGNORECASE):
-                    log_entry_commit = False
+                    if match_keyword.group(1).lower() == "Title".lower():
```
Comment (kacf) [inline extra/changelog-generator/changelog-generator:170]: The second `lower()` is very overkill, why not just type `title`? Same for the others.

File: extra/changelog-generator/changelog-generator (hunk 1 above)
Comment (lluiscampos) [reply extra/changelog-generator/changelog-generator:158]: Yes, I see what you mean, thanks.

I will replace it with: `^(Title|Commit|All|None)[ .:]+(.*)$`

dependabot uses something like:
````
Changelog:All : Bump bla bla bla
```

Comment (lluiscampos) [reply extra/changelog-generator/changelog-generator]: @kacf It took me a while to fine tune this regex... I added more test cases that I hope cover the relevant corner cases.

File: extra/changelog-generator/changelog-generator [hunk 3]
```diff
@@ -148,14 +148,20 @@ for repo in repos:
...
                     add_entry(sha, log_entry)
                     log_entry = ""
                 log_entry_local = False
 
-                if re.match("^(Title|Commit|None)[ .]*$", match.group(1), re.IGNORECASE):
+                # The logic of this regex is:
+                # - Shall start with Title|Commit|All|None
+                # - From here, if [.:]+ then capture the rest in a new group
+                # - Else just expect end line
+                match_keyword = re.match("^(Title|Commit|All|None) *([.:]+)? *((?(2).*$|$))", match_changelog.group(1), re.IGNORECASE)
```
Comment (kacf) [inline extra/changelog-generator/changelog-generator:162]: I actually have not seen the `(?(` notation before, I had to look it up.

Question is, is this an easier way to express the same thing?

`^(Title|Commit|All|None) *(?:[.:]+ *(.*)$|$))`

If it doesn't work, I'm fine with the original expression.

File: extra/changelog-generator/changelog-generator [hunk 4]
```diff
@@ -166,15 +172,20 @@ for repo in repos:
...
-                    log_entry_commit = True
-                elif re.match("^None[ .]*$", match.group(1), re.IGNORECASE):
-                    log_entry_commit = False
+                    if match_keyword.group(1).lower() == "title":
+                        log_entry = title
+                    elif match_keyword.group(1).lower() == "none":
+                        log_entry_commit = False
+                    elif match_keyword.group(1).lower() in ["commit", "all"]:
+                        log_entry_commit = True
+                        if match_keyword.group(3):
```
Comment (kacf) [inline extra/changelog-generator/changelog-generator:181]: If you change the expression above, this will be group 2.

File: extra/changelog-generator/changelog-generator [hunk 5]
```diff
@@ -166,15 +172,20 @@ for repo in repos:
...
+                            # Log the rest of the line
+                            if commit_msg:
+                                commit_msg += "\n"
+                            commit_msg += match_keyword.group(3)
```
Comment (kacf) [inline extra/changelog-generator/changelog-generator:185]: And here.

File: extra/changelog-generator/changelog-generator (hunk 3 above)
Comment (lluiscampos) [reply extra/changelog-generator/changelog-generator]: Your regex also works (all tests pass, which is all the evidence I need). Thank you!
